packages/api/src/controllers/managementPlanController.js

```diff
@@ -14,23 +14,109 @@
  */
 
 import ManagementPlanModel from '../models/managementPlanModel.js';
+import ManagementPlanGroup from '../models/managementPlanGroupModel.js';
 import CropManagementPlanModel from '../models/cropManagementPlanModel.js';
 import ManagementTasksModel from '../models/managementTasksModel.js';
 import TaskModel from '../models/taskModel.js';
 import TaskTypeModel from '../models/taskTypeModel.js';
 import FieldWorkTypeModel from '../models/fieldWorkTypeModel.js';
 import TransplantTaskModel from '../models/transplantTaskModel.js';
 import PlantTaskModel from '../models/plantTaskModel.js';
-import { raw } from 'objection';
-import lodash from 'lodash';
+import UserFarmModel from '../models/userFarmModel.js';
+import objection, { raw } from 'objection';
+import _pick from 'lodash/pick.js';
 import { sendTaskNotification, TaskNotificationTypes } from './taskController.js';
+import {
+  getDatesFromManagementPlanGraph,
+  getManagementPlanGroupTemplateGraph,
+  getFormattedManagementPlanData,
+} from '../util/copyCropPlan.js';
+import { getSortedDates } from '../util/util.js';
+const { transaction, Model } = objection;
 
 const managementPlanController = {
+  repeatManagementPlan() {
+    return async (req, res) => {
+      const trx = await transaction.start(Model.knex());
+      const { startDates, management_plan_id, repeatDetails } = req.body;
```

Comment:
There is a mixture of `snake_case` and `camelCase` in the property names, we should try to decide on casing.
e.g.
```
src/controllers/sensorController.js
534:      const { start_time, end_time } = req.body;
548:      const { locationIds = [], readingTypes = [], endUnixTime = 0, startUnixTime = 0 } = req.body;
589:      const { location_id } = req.body;
```
If we decide on `snake_case`, we should change to `start_dates`, `repeat_details`, `management_plan_id`

Reply:
I can set the rest to snake case I think I was basing it off an earlier discussion. If @SayakaOno and @kathyavini can adjust to snake case on their parts as needed too?

Reply:
I don't think my part is relevant as it goes through LF-3378, so it's all good by me 👍

Reply:
now in snake case